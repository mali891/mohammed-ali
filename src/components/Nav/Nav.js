import React from "react"
import { Link } from "gatsby"

import { styles } from "./Nav.styles"
import Logo5 from "../../img/Logo5"
import { URLS } from "../../constants"

const Nav = () => (
  <nav css={styles} className="c-nav">
    {
      <Link className="c-link__no-hover" to={URLS.HOME}>
        <Logo5 />
      </Link>
    }
    <ul>
      <li>
        <Link to={URLS.HOME}>Home</Link>
      </li>
      <li>
        <Link to={URLS.REVIEWS}>Reviews</Link>
      </li>
      <li>
        <Link to={URLS.BLOG}>Blog</Link>
      </li>
      <li>
        <Link to={URLS.CONTACT}>Contact</Link>
      </li>
    </ul>
  </nav>
)

export default Nav
