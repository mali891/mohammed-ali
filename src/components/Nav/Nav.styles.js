import { css } from '@emotion/core'
import { FONT_SIZES, SPACING, MEDIA_QUERIES } from '../../styles/settings'

export const styles = css`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100vw;
  height: ${SPACING.MOBILE.XL};
  padding: 0 ${SPACING.MOBILE.SM};
  z-index: 1;

  .c-nav__overlay {
    height: 100vh;
    width: 100vw;
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background: rgba(0, 0, 0, 0.6);
    z-index: 1;
  }

  .c-nav__logo {
    width: 5rem;
  }

  .c-nav__list {
    margin-bottom: ${SPACING.MOBILE.XL};
  }

  .c-nav__list-item {
    display: block;
    margin-bottom: ${SPACING.MD};
    background-color: none;
  }

  .c-nav__link {
    font-size: ${FONT_SIZES.SM};
  }

  .c-nav__menu {
    display: flex;
    align-items: flex-start;
    flex-direction: column;
    position: absolute;
    height: 100vh;
    width: 80%;
    max-width: 40rem;
    top: 0;
    right: 0;
    bottom: 0;
    padding: ${SPACING.MOBILE.XL} ${SPACING.MOBILE.SM} ${SPACING.MOBILE.XL};
    background-color: var(--colour-tertiary);
    transform: translateX(100%);
    z-index: 2;

    &.c-nav__menu--is-expanded {
      transform: translateX(0);
    }
  }

  @media (min-width: ${MEDIA_QUERIES.NOTEBOOK}) {
    height: ${SPACING.LG};
    padding: 0 ${SPACING.MD};

    .c-nav__logo {
      width: 7.5rem;
    }

    .c-nav__menu {
      flex-direction: row;
      align-items: center;
      border-left: none;
      position: relative;
      height: unset;
      width: unset;
      max-width: unset;
      top: unset;
      right: unset;
      bottom: unset;
      background-color: transparent;
      transform: translateX(0);
      padding: 0;
    }

    .c-nav__list {
      margin-right: 5rem;
      margin-bottom: unset;
    }

    .c-nav__list-item {
      display: inline;
      margin-left: ${SPACING.MD};
      background-color: none;
    }

    .c-nav__link {
      font-size: ${FONT_SIZES.SM};
    }
  }
`
