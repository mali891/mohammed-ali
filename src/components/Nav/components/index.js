import Hamburger from './Hamburger/Hamburger'
import ThemeToggle from './ThemeToggle/ThemeToggle'

export { Hamburger, ThemeToggle }
