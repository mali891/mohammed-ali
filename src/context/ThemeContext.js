import React, { createContext, useState } from 'react'
import PropTypes from 'prop-types'

const { Provider, Consumer } = createContext()

const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('dark')

  return <Provider value={{ theme, setTheme }}>{children}</Provider>
}

ThemeProvider.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
}

export { ThemeProvider, Consumer as ThemeConsumer }
