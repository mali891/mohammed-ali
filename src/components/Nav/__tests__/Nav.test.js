import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Nav from '../Nav'
import { ThemeProvider } from '../../../context/ThemeContext'

describe('Nav', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Nav footerRef={{ current: {} }} />
      </ThemeProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking the hamburger activates the menu', () => {
    const { asFragment, container } = render(
      <ThemeProvider>
        <Nav footerRef={{ current: {} }} />
      </ThemeProvider>
    )
    const hamburger = container.querySelector('.c-hamburger')

    expect(container.querySelector('.c-nav__overlay')).not.toBeInTheDocument()
    expect(container.querySelector('.c-nav__menu--is-expanded')).not.toBeInTheDocument()

    fireEvent.click(hamburger)

    expect(container.querySelector('.c-nav__overlay')).toBeInTheDocument()
    expect(container.querySelector('.c-nav__menu--is-expanded')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
