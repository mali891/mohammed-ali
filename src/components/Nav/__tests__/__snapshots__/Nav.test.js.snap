// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav clicking the hamburger activates the menu 1`] = `
<DocumentFragment>
  <nav
    class="c-nav u-bg--primary u-clr--secondary"
    css="[object Object]"
    role="navigation"
  >
    <div
      class="c-nav__overlay u-anim--fade-in"
      role="presentation"
    />
    <a
      aria-label="Back to home page"
      class="c-link u-clr--secondary c-link--no-hover"
      css="[object Object]"
      href="/"
    >
      <svg
        class="c-nav__logo"
        viewBox="0 0 400 400"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="linear-gradient-1"
            x1="53.83"
            x2="246.17"
            y1="187.54"
            y2="187.54"
          >
            <stop
              offset="0"
              stop-color="#8fd3f4"
            />
            <stop
              offset=".11"
              stop-color="#7dd8e8"
            />
            <stop
              offset=".13"
              stop-color="#7ad9e6"
              stop-opacity=".97"
            />
            <stop
              offset=".26"
              stop-color="#6cdddd"
              stop-opacity=".8"
            />
            <stop
              offset=".33"
              stop-color="#5de1d3"
              stop-opacity=".86"
            />
            <stop
              offset=".46"
              stop-color="#46e8c4"
              stop-opacity=".96"
            />
            <stop
              offset=".53"
              stop-color="#3deabe"
            />
            <stop
              offset=".55"
              stop-color="#3ceabe"
            />
            <stop
              offset=".83"
              stop-color="#37ecba"
            />
          </lineargradient>
        </defs>
        <g>
          <g>
            <path
              d="M205 193.66l-10-17.32-40-69.28c-2.75-4.77-7.25-4.77-10 0l-40 69.28-10 17.32-40 69.28c-2.75 4.77-.5 8.66 5 8.66h180c5.5 0 7.75-3.89 5-8.66zm-125 66c-5.5 0-7.75-3.9-5-8.66l30-52 10-17.32 30-52c2.75-4.77 7.25-4.77 10 0l30 52L195 199l30 52c2.75 4.76.5 8.66-5 8.66H80z"
              fill="url(#linear-gradient-1)"
            />
            <path
              class="u-fill--secondary"
              d="M305 193.66l-10-17.32-40-69.28c-2.75-4.77-7.25-4.77-10 0l-40 69.28-10 17.32-40 69.28c-2.75 4.77-.5 8.66 5 8.66h180c5.5 0 7.75-3.89 5-8.66zm-125 66c-5.5 0-7.75-3.9-5-8.66l30-52 10-17.32 30-52c2.75-4.77 7.25-4.77 10 0l30 52L295 199l30 52c2.75 4.76.5 8.66-5 8.66H180z"
            />
            <path
              class="u-fill--secondary"
              d="M200 0C89.54 0 0 89.54 0 200s89.54 200 200 200 200-89.54 200-200S310.46 0 200 0zm0 387.5C96.45 387.5 12.5 303.55 12.5 200S96.45 12.5 200 12.5s187.5 84 187.5 187.5S303.55 387.5 200 387.5z"
            />
          </g>
        </g>
      </svg>
    </a>
    <div
      class="c-nav__menu u-bg--primary c-nav__menu--is-expanded"
      id="navigation"
    >
      <ul
        class="c-nav__list"
      >
        <li
          class="c-nav__list-item"
        >
          <a
            class="c-nav__link c-link u-clr--secondary"
            css="[object Object]"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="c-nav__list-item"
        >
          <a
            class="c-nav__link c-link u-clr--secondary"
            css="[object Object]"
            href="/reviews"
          >
            Reviews
          </a>
        </li>
        <li
          class="c-nav__list-item"
        >
          <a
            class="c-nav__link c-link u-clr--secondary"
            css="[object Object]"
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li
          class="c-nav__list-item"
        >
          <button
            class="c-nav__link c-link u-clr--secondary"
            css="[object Object]"
          >
            Contact
          </button>
        </li>
      </ul>
      <div
        aria-label="Toggle theme and colour contrast"
        class="c-theme-toggle c-theme-toggle--dark"
        css="[object Object]"
      >
        <input
          class="c-theme-toggle__input"
          id="theme-toggle"
          readonly=""
          type="checkbox"
        />
        <label
          class="c-theme-toggle__label"
          for="theme-toggle"
          tabindex="0"
        >
          <div
            class="c-theme-toggle__toggle"
          >
            <svg
              class="u-anim--fade-in icon--moon c-theme-toggle__icon"
              viewBox="0 0 487.42 513.3"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M282.89.8a240.56 240.56 0 0158.41 157.42c0 133.44-108.17 241.61-241.61 241.61a240.59 240.59 0 01-98.53-21A258.81 258.81 0 00228 512.8c143 0 259-115.93 259-258.94C486.92 129.7 399.5 26 282.89.8zM228 494a240.28 240.28 0 01-179.16-80.29 260.85 260.85 0 0050.85 5c143.61 0 260.45-116.84 260.45-260.45A260.36 260.36 0 00329.9 36.52c82.45 38.93 138.17 123.38 138.17 217.34C468.07 386.25 360.36 494 228 494z"
                data-name="moon"
                fill="#FFE29F"
                stroke="#FFE29F"
                stroke-miterlimit="10"
                stroke-width="3px"
              />
            </svg>
          </div>
        </label>
      </div>
    </div>
    <button
      aria-controls="navigation"
      aria-expanded="true"
      aria-label="Menu"
      class="c-hamburger c-hamburger--collapse c-hamburger--is-active"
      css="[object Object]"
      type="button"
    >
      <span
        class="c-hamburger__box"
      >
        <span
          class="c-hamburger__inner u-bg--secondary"
        />
      </span>
    </button>
  </nav>
</DocumentFragment>
`;

exports[`Nav renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="c-nav u-bg--primary u-clr--secondary"
    css="[object Object]"
    role="navigation"
  >
    <a
      aria-label="Back to home page"
      class="c-link u-clr--secondary c-link--no-hover"
      css="[object Object]"
      href="/"
    >
      <svg
        class="c-nav__logo"
        viewBox="0 0 400 400"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="linear-gradient-1"
            x1="53.83"
            x2="246.17"
            y1="187.54"
            y2="187.54"
          >
            <stop
              offset="0"
              stop-color="#8fd3f4"
            />
            <stop
              offset=".11"
              stop-color="#7dd8e8"
            />
            <stop
              offset=".13"
              stop-color="#7ad9e6"
              stop-opacity=".97"
            />
            <stop
              offset=".26"
              stop-color="#6cdddd"
              stop-opacity=".8"
            />
            <stop
              offset=".33"
              stop-color="#5de1d3"
              stop-opacity=".86"
            />
            <stop
              offset=".46"
              stop-color="#46e8c4"
              stop-opacity=".96"
            />
            <stop
              offset=".53"
              stop-color="#3deabe"
            />
            <stop
              offset=".55"
              stop-color="#3ceabe"
            />
            <stop
              offset=".83"
              stop-color="#37ecba"
            />
          </lineargradient>
        </defs>
        <g>
          <g>
            <path
              d="M205 193.66l-10-17.32-40-69.28c-2.75-4.77-7.25-4.77-10 0l-40 69.28-10 17.32-40 69.28c-2.75 4.77-.5 8.66 5 8.66h180c5.5 0 7.75-3.89 5-8.66zm-125 66c-5.5 0-7.75-3.9-5-8.66l30-52 10-17.32 30-52c2.75-4.77 7.25-4.77 10 0l30 52L195 199l30 52c2.75 4.76.5 8.66-5 8.66H80z"
              fill="url(#linear-gradient-1)"
            />
            <path
              class="u-fill--secondary"
              d="M305 193.66l-10-17.32-40-69.28c-2.75-4.77-7.25-4.77-10 0l-40 69.28-10 17.32-40 69.28c-2.75 4.77-.5 8.66 5 8.66h180c5.5 0 7.75-3.89 5-8.66zm-125 66c-5.5 0-7.75-3.9-5-8.66l30-52 10-17.32 30-52c2.75-4.77 7.25-4.77 10 0l30 52L295 199l30 52c2.75 4.76.5 8.66-5 8.66H180z"
            />
            <path
              class="u-fill--secondary"
              d="M200 0C89.54 0 0 89.54 0 200s89.54 200 200 200 200-89.54 200-200S310.46 0 200 0zm0 387.5C96.45 387.5 12.5 303.55 12.5 200S96.45 12.5 200 12.5s187.5 84 187.5 187.5S303.55 387.5 200 387.5z"
            />
          </g>
        </g>
      </svg>
    </a>
    <div
      class="c-nav__menu u-bg--primary"
      id="navigation"
    >
      <ul
        class="c-nav__list"
      >
        <li
          class="c-nav__list-item"
        >
          <a
            class="c-nav__link c-link u-clr--secondary"
            css="[object Object]"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="c-nav__list-item"
        >
          <a
            class="c-nav__link c-link u-clr--secondary"
            css="[object Object]"
            href="/reviews"
          >
            Reviews
          </a>
        </li>
        <li
          class="c-nav__list-item"
        >
          <a
            class="c-nav__link c-link u-clr--secondary"
            css="[object Object]"
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li
          class="c-nav__list-item"
        >
          <button
            class="c-nav__link c-link u-clr--secondary"
            css="[object Object]"
          >
            Contact
          </button>
        </li>
      </ul>
      <div
        aria-label="Toggle theme and colour contrast"
        class="c-theme-toggle c-theme-toggle--dark"
        css="[object Object]"
      >
        <input
          class="c-theme-toggle__input"
          id="theme-toggle"
          readonly=""
          type="checkbox"
        />
        <label
          class="c-theme-toggle__label"
          for="theme-toggle"
          tabindex="0"
        >
          <div
            class="c-theme-toggle__toggle"
          >
            <svg
              class="u-anim--fade-in icon--moon c-theme-toggle__icon"
              viewBox="0 0 487.42 513.3"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M282.89.8a240.56 240.56 0 0158.41 157.42c0 133.44-108.17 241.61-241.61 241.61a240.59 240.59 0 01-98.53-21A258.81 258.81 0 00228 512.8c143 0 259-115.93 259-258.94C486.92 129.7 399.5 26 282.89.8zM228 494a240.28 240.28 0 01-179.16-80.29 260.85 260.85 0 0050.85 5c143.61 0 260.45-116.84 260.45-260.45A260.36 260.36 0 00329.9 36.52c82.45 38.93 138.17 123.38 138.17 217.34C468.07 386.25 360.36 494 228 494z"
                data-name="moon"
                fill="#FFE29F"
                stroke="#FFE29F"
                stroke-miterlimit="10"
                stroke-width="3px"
              />
            </svg>
          </div>
        </label>
      </div>
    </div>
    <button
      aria-controls="navigation"
      aria-expanded="false"
      aria-label="Menu"
      class="c-hamburger c-hamburger--collapse"
      css="[object Object]"
      type="button"
    >
      <span
        class="c-hamburger__box"
      >
        <span
          class="c-hamburger__inner u-bg--secondary"
        />
      </span>
    </button>
  </nav>
</DocumentFragment>
`;
